### PR TITLE
WIP: Add signal channel without app interface

### PR DESCRIPTION
**Pull request by Connoropolous**
Ready aside from writing a test. Request for comment.

I have a project where I've experimented with running my own conductor via Rust, and Holochain imported as a library instead of the binary, and I wanted to listen to signals, without attaching any ports/interfaces ... that led to this. 

I can try to add a test if you like it.

### CODE REVIEW

File: crates/holochain/src/conductor/interface.rs [hunk 1]
```diff
@@ -31,6 +32,11 @@ impl SignalBroadcaster {
         Ok(())
     }
 
+    /// get the inner vec of senders
+    pub fn into_inner(&self) -> Vec<broadcast::Sender<Signal>> {
+        self.0.clone()
+    }
```
Comment (Connoropolous) [inline crates/holochain/src/conductor/interface.rs:38]: without this we can't retrieve the Vec of sender for the sake of attaching new listeners either. so it's very limiting.

File: crates/holochain/src/conductor/conductor.rs [hunk 2]
```diff
@@ -311,16 +314,26 @@ where
...
     }
 
+    pub(super) fn add_signal_channel(
+        &mut self,
+    ) -> ConductorResult<tokio::sync::broadcast::Receiver<Signal>> {
+        let (signal_broadcaster, signal_receiver) =
+            tokio::sync::broadcast::channel(SIGNAL_BUFFER_SIZE);
+        // no associated AppInterfaceId
+        self.signal_broadcasters.push((None, signal_broadcaster));
+        Ok(signal_receiver)
```
Comment (Connoropolous) [inline crates/holochain/src/conductor/conductor.rs:329]: return the signal_receiver so that the caller can listen to the signals stream, which is prbly what they want to do (I did)

File: crates/holochain/src/conductor/interface.rs (hunk 1 above)
Comment (maackle) [reply crates/holochain/src/conductor/interface.rs:38]: @Connoropolous do you need individual access to each sender, or does it perhaps make sense to provide a method to listen to all of them at once instead?

Comment (Connoropolous) [reply crates/holochain/src/conductor/interface.rs:38]: soo... technically we could leave this bit out for now, since I don't need it. I thought it was funny that there would be no way to start listening, if you have one of these SignalBroadcaster structs...
I don't know what the ultimate needs for this would be

Comment (maackle) [reply crates/holochain/src/conductor/interface.rs:38]: Yeah I'd prefer to not give access to the internals if we don't know what the use case will be.

**Comment by maackle** [comment]
@Connoropolous I'm on the fence about merging this. Our current interface implementation is only partial, in that we broadcast all signals to all interfaces, so there is no association between interfaces and apps, which there may be in the future. Moreover, we want to associate an interface with an agent in the future. These changes might have an impact on the proposed `Option<AppInterfaceId>` pattern here.

Basically, I don't know if we can guarantee that this new API will be supported in the future. I guess it will be fine if we say that signal listeners added this way will always receive all signals on all interfaces. 

If that's the case though, what if we put this behind a feature flag that's not on by default? I wouldn't want this to wind up getting used in the actual Holochain binary.

**Comment by Connoropolous** [comment]
Ima just close it I'm not worried about it.
